### PR TITLE
[posix] add posix support for sending RA messages to routing manager

### PR DESCRIPTION
**Pull request by erjiaqing**

### CODE REVIEW

File: src/posix/platform/netif.cpp [hunk 1]
```diff
@@ -1021,6 +1021,70 @@ static void processReceive(otMessage *aMessage, void *aContext)
     }
 }
 
+#if OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE || OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
```
Comment (abtink) [inline src/posix/platform/netif.cpp:1024]: I may be just not seeing it, but where do we `#endif` this?

File: src/posix/platform/netif.cpp [hunk 2]
```diff
@@ -1021,6 +1021,70 @@ static void processReceive(otMessage *aMessage, void *aContext)
...
+exit:
+    return ret;
+}
+
+/**
+ * Returns false if the message is not an ICMPv6 RA message.
+ *
+ */
+static bool tryProcessIcmp6RaMessage(otInstance *aInstance, const uint8_t *data, ssize_t length)
+{
```
Comment (abtink) [inline src/posix/platform/netif.cpp:1068]: Return `otError`  instead of `bool`?

**Review by abtink — COMMENTED** [review]
Thanks @erjiaqing Looks good. 
Just some smaller suggestions below.

File: src/posix/platform/netif.cpp (hunk 1 above)
Comment (erjiaqing) [reply src/posix/platform/netif.cpp:1024]: after `getIpVersion`

Comment (abtink) [reply src/posix/platform/netif.cpp:1025]: Thanks 😄

**Review by abtink — APPROVED** [review]
Thanks @erjiaqing. LGTM. 👍

**Comment by erjiaqing** [comment]
@jwhui PTAL thanks.

File: src/posix/platform/netif.cpp [hunk 3]
```diff
@@ -1021,6 +1022,71 @@ static void processReceive(otMessage *aMessage, void *aContext)
...
+    otIcmp6Header  icmpHeader;
+
+    VerifyOrExit(length >= OT_IP6_HEADER_SIZE + OT_ICMP6_ROUTER_ADVERT_MIN_SIZE);
+    VerifyOrExit(getIpVersion(data) == kIpVersion6);
+    VerifyOrExit(data[OT_IP6_HEADER_PROTO_OFFSET] == OT_IP6_PROTO_ICMP6);
+
+    ret = data + OT_IP6_HEADER_SIZE;
+    memcpy(&icmpHeader, ret, sizeof(icmpHeader));
+    VerifyOrExit(icmpHeader.mType == OT_ICMP6_TYPE_ROUTER_ADVERT);
+    VerifyOrExit(icmpHeader.mCode == 0);
```
Comment (jwhui) [inline src/posix/platform/netif.cpp:1059]: Return `nullptr` if check fails?

File: include/openthread/instance.h [hunk 4]
```diff
@@ -53,7 +53,7 @@ extern "C" {
  * @note This number versions both OpenThread platform and user APIs.
  *
  */
-#define OPENTHREAD_API_VERSION (331)
+#define OPENTHREAD_API_VERSION (332)
```
Comment (jwhui) [inline include/openthread/instance.h:56]: Rebase and increment.

File: include/openthread/icmp6.h [hunk 5]
```diff
@@ -80,7 +80,8 @@ typedef enum otIcmp6Code
     OT_ICMP6_CODE_FRAGM_REAS_TIME_EX   = 1, ///< Fragment Reassembly Time Exceeded
 } otIcmp6Code;
 
-#define OT_ICMP6_HEADER_DATA_SIZE 4 ///< Size of an message specific data of ICMPv6 Header.
+#define OT_ICMP6_HEADER_DATA_SIZE 4        ///< Size of an message specific data of ICMPv6 Header.
+#define OT_ICMP6_ROUTER_ADVERT_MIN_SIZE 16 ///< Size of an Router Advertisement message without any options.
```
Comment (jwhui) [inline include/openthread/icmp6.h:84]:
```suggestion
#define OT_ICMP6_HEADER_DATA_SIZE 4        ///< Size of ICMPv6 Header.
#define OT_ICMP6_ROUTER_ADVERT_MIN_SIZE 16 ///< Size of a Router Advertisement message without any options.
```

File: src/posix/platform/netif.cpp [hunk 6]
```diff
@@ -1021,6 +1022,71 @@ static void processReceive(otMessage *aMessage, void *aContext)
...
+    OT_UNUSED_VARIABLE(kIpVersion6);
+
+    return (static_cast<uint8_t>(data[0]) >> 4) & 0x0F;
+}
+#endif
+
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
+
+/**
+ * Returns nullptr if data is not pointed to a valid ICMPv6 RA message.
```
Comment (jwhui) [inline src/posix/platform/netif.cpp:1044]:
```suggestion
 * Returns nullptr if data does not point to a valid ICMPv6 RA message.
```

File: src/posix/platform/netif.cpp (hunk 3 above)
Comment (erjiaqing) [reply src/posix/platform/netif.cpp]: Done

**Review by jwhui — APPROVED** [review]

**Pull request merged**